apps/src/blockly/addons/cdoUtils.js

```diff
@@ -106,9 +108,167 @@ export function getUserTheme(themeOption) {
 }
 
 export function getCode(workspace) {
+  // Begin collecting data about our ability to correctly serialize a workspace in JSON format.
+  if (
+    experiments.isEnabled(experiments.BLOCKLY_JSON) ||
+    Math.random() <= 0.05
+  ) {
+    testJsonSerialization(workspace);
```

Comment:
Can we do this comparison asynchronously?

Reply:
Not really, unless we find a way to also spin up an instance of the Blockly lab with the appropriate custom blocks installed. Realistically, we'd need the actual curriculum level hosts the project to be sure that we're able to create all of the blocks.

Reply:
I just mean can we do this in the background (and continue to load the page)? As written, this will block actually loading the workspace, yeah?

Reply:
Sorry for misunderstanding. The test is now being run asynchronously. Great suggestion!